rom/dev/src/flow/cold_reset/fw_processor.rs

```diff
@@ -735,4 +809,78 @@ impl FirmwareProcessor {
 
         Ok(())
     }
+
+    /// Retrieve the fw image from the recovery interface and store it in the mailbox sram.
+    ///
+    /// # Arguments
+    /// * `dma` - DMA driver
+    /// * `soc_ifc` - SOC Interface
+    ///
+    /// # Returns
+    /// * `()` - Ok
+    ///   Error code on failure.
+    fn retrieve_image_from_recovery_interface(
+        dma: &mut Dma,
+        soc_ifc: &mut SocIfc,
+    ) -> CaliptraResult<u32> {
+        let rri_base_addr: u64 = soc_ifc.recovery_interface_base_addr();
```

Comment:
I'm confused. I thought the recovery interface was accessible via the I3C registers (so didn't need DMA): https://github.com/chipsalliance/caliptra-sw/blob/main-2.x/hw/latest/registers/src/i3ccsr.rs#L624

Reply:
Nevermind, we discussed this, and it sounds like this will only be accessible via DMA.